File: backend/src/main/java/com/hardware/erp/expense/service/ExpenseCategoryService.java
package com.hardware.erp.expense.service;

import com.hardware.erp.expense.dto.ExpenseCategoryRequest;
import com.hardware.erp.expense.dto.ExpenseCategoryResponse;

import java.util.List;

public interface ExpenseCategoryService {

    List<ExpenseCategoryResponse> list();

    ExpenseCategoryResponse create(ExpenseCategoryRequest request);

    ExpenseCategoryResponse update(Long id, ExpenseCategoryRequest request);
}

File: frontend/src/modules/expense/services/expenseCategoryService.ts
import { apiGet, apiPost, apiPut } from '@/services/apiClient';
import type { ExpenseCategoryRequest, ExpenseCategoryResponse } from '../types';

/** Backend: expense/controller/ExpenseCategoryController.java */
export const expenseCategoryService = {
  list: () => apiGet<ExpenseCategoryResponse[]>('/v1/expense-categories'),

  create: (body: ExpenseCategoryRequest) =>
    apiPost<ExpenseCategoryResponse>('/v1/expense-categories', body),

  update: (id: number, body: ExpenseCategoryRequest) =>
    apiPut<ExpenseCategoryResponse>(`/v1/expense-categories/${id}`, body),
};
